src/app/shared/services/broker.service.ts

```diff
@@ -107,7 +107,13 @@ export class BrokerService {
   parseErrorBlob(err: HttpErrorResponse): Observable<any> {
     const reader: FileReader = new FileReader();
     reader.readAsText(err.error);
-    return throwError(JSON.parse(reader.result as string));
+    const obs = new Observable((observer: any) => {
```

Comment:
why observable and not promise? this is producing a single value result if I am not wrong.

Reply:
Because this function needs to return observable and is being piped and subscribed at by the caller. Also, I personally would prefer observables over promises just for consistency in our app.